### ISSUE TITLE
Hook LuckPerms

### ISSUE THREAD

**Issue by DesperoKF** [sgtcaze/NametagEdit#360]
# Bug Report

**Type /ver NametagEdit and post the output.**
NametagEdit version 4.4.2
Authors: sgtcaze and Cory

**What version of Bukkit/Spigot/PaperSpigot are you using? Type /ver**
 This server is running Paper version git-Paper-1613 (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT)

**What plugins are you using? Type /plugins**

Plugins(61): AntiLagS, BetterChairs, BKCommonLib, Broadcaster, BungeeSystem, ChatControl, Citizens, ClearLag, Crates, CustomEnderChest, CustomPayloadFixer, CustomTab, Essentials, EssentialsChat, EssentialsSpawn, ExploitFixer, FastAsyncWorldEdit, FreundeGui, GG_Features, GG_Prefix, GGVoteListener, Glowing, HeadDatabase, HideAchievements, HolographicDisplays, iDisguise, IllegalStack, ItemSignatur, LagAssist, LightCleaner, LuckPerms, LuckPerms-GUI, MergedMob, Multiverse-Core, MysqlPlayerDataBridge, NametagEdit, NoCheatPlus, Orebfuscator4, PlaceholderAPI, PlotSquared, PlugMan, PremiumVanish, ProtocolLib, redclockdetector, ServerlistMOTD, SignEdit, SilkSpawners, SimpleAPI, SimplePets, SkinChanger, SuperTrails, ValuableItemsScanner, Vault, ViaBackwards, ViaRewind, ViaVersion, VillagerShop, Votifier, Walls, WorldEdit, WorldGuard

**Please explain your issue. How do you trigger it?**
I have made this in the groups.yml
Groups:
  Owner:
    Permission: group.owner
    Prefix: '&4&lOwner : '
    SortPriority: 1
  Admin:
    Permission: group.admin
    Prefix: '&c&lAdmin : '
    SortPriority: 2

But when i change ingame the Prefix. Its update on Chat & Tablist but nothing in the Name over Head. 

I have the PlaceHolderAPI (LuckPerms & mvdw Extensions) 

Code: https://hastebin.com/qagubunowo.java

Screen: http://prntscr.com/npkyxk

**Are there any errors in the console? Please use: https://pastebin.com**
No Errors in the Console!

**Comment by gelsbern** [comment]
Are you using group inheritance in Luckperms?  I had this issue myself.  First I had to set up a default group for all players.  I called it citizen and set the prefix to &f (white).  Then I created my other groups with the proper prefixes.  So then in Luckperms I set it that all users get the nte.citizens permissions

**Citizen**

in the namtagedit config.yml:

Groups:
citizen:
Permission: group.owner
Prefix: '&f'
SortPriority: 1
...

Then in Luckperms: 
/lp group citizen set permission nte.citizen true

The problem here is that the display priority in the config is only for the tab list, and if you do things properly with a permissions plugins, the next rank (admin) inherits all the permissions from the lower ranks.   So in order for each rank to have their proper name color I had to negate the permissions from the lower ranks,  Like this:

**Admin**
/lp group admin set permission nte.citizen false
/lp group admin set permission nte.admin true

**Owner**
/lp group owner set permission nte.citizen false
/lp group owner set permission nte.admin false
/lp group owner set permission nte.owner true

This ended up working correctly.  I had to do some trickery with a couple of plugins to (PremiumVanish) to issue  /nte reload when those who are allowed to go into vanish become visible again, otherwise their name stays white. 

Hope this helpt. 
Once I did this my admin and owner accounts all had the proper colored names.
